internal/controller/stas/scan_job_controller.go

```diff
@@ -202,28 +201,20 @@ func (r *ScanJobReconciler) reconcileCompleteJob(ctx context.Context, job *batch
 	cis.Status.LastScanJobUID = job.UID
 	cis.Status.LastSuccessfulScanTime = &now
 
-	err = r.Status().Patch(ctx, cis, client.MergeFrom(cleanCis))
-	if err != nil && isResourceTooLargeError(err) {
-		cis = cleanCis.DeepCopy()
-
-		condition := metav1.Condition{
-			Type:    string(kstatus.ConditionStalled),
-			Status:  metav1.ConditionTrue,
-			Reason:  stasv1alpha1.ReasonVulnerabilityOverflow,
-			Message: fmt.Sprintf("vulnerability report is too large to fit in API: %s", err),
+	for {
```

Comment:
Exit condition is quite hidden without it
```suggestion
	for minSeverity < stasv1alpha1.MaxSeverity {
```

Reply:
or 
```suggestion
	for filterSeverity := minSeverity ; i < stasv1alpha1.MaxSeverity;  filterSeverity++ {
```

Reply:
Fixed